ballerina/client.bal

```diff
@@ -12,4 +12,80 @@
+public isolated client class Client {
+    final http:Client clientEp;
+    # Gets invoked to initialize the `connector`.
+    #
+    # + config - The configurations to be used when initializing the `connector` 
+    # + serviceUrl - URL of the target service 
+    # + return - An error if connector initialization failed 
+    public isolated function init(ConnectionConfig config, string serviceUrl = "https://api.openai.com/v1") returns error? {
+        http:ClientConfiguration httpClientConfig = {auth: config.auth, httpVersion: config.httpVersion, timeout: config.timeout, forwarded: config.forwarded, poolConfig: config.poolConfig, compression: config.compression, circuitBreaker: config.circuitBreaker, retryConfig: config.retryConfig, validation: config.validation};
+        do {
+            if config.http1Settings is ClientHttp1Settings {
+                ClientHttp1Settings settings = check config.http1Settings.ensureType(ClientHttp1Settings);
+                httpClientConfig.http1Settings = {...settings};
+            }
+            if config.http2Settings is http:ClientHttp2Settings {
+                httpClientConfig.http2Settings = check config.http2Settings.ensureType(http:ClientHttp2Settings);
+            }
+            if config.cache is http:CacheConfig {
+                httpClientConfig.cache = check config.cache.ensureType(http:CacheConfig);
+            }
+            if config.responseLimits is http:ResponseLimitConfigs {
+                httpClientConfig.responseLimits = check config.responseLimits.ensureType(http:ResponseLimitConfigs);
+            }
+            if config.secureSocket is http:ClientSecureSocket {
+                httpClientConfig.secureSocket = check config.secureSocket.ensureType(http:ClientSecureSocket);
+            }
+            if config.proxy is http:ProxyConfig {
+                httpClientConfig.proxy = check config.proxy.ensureType(http:ProxyConfig);
+            }
+        }
+        http:Client httpEp = check new (serviceUrl, httpClientConfig);
+        self.clientEp = httpEp;
+        return;
+    }
+
+    # Generates audio from the input text.
+    #
+    # + headers - Headers to be sent with the request 
+    # + return - OK 
+    resource isolated function post audio/speech(CreateSpeechRequest payload, map<string|string[]> headers = {}) returns byte[]|error {
+        string resourcePath = string `/audio/speech`;
+        http:Request request = new;
+        json jsonBody = payload.toJson();
+        request.setPayload(jsonBody, "application/json");
+        return self.clientEp->post(resourcePath, request, headers);
+    }
+
+    # Transcribes audio into the input language.
+    #
+    # + headers - Headers to be sent with the request 
+    # + return - OK 
+    resource isolated function post audio/transcriptions(CreateTranscriptionRequest payload, map<string|string[]> headers = {}) returns inline_response_200|error {
```

Comment:
@TharmiganK @lnash94 Here the counterpart contract section looks like below.

<img width="723" alt="Screenshot 2024-08-08 at 10 39 09" src="https://github.com/user-attachments/assets/a002c3d7-c9f3-453c-bee1-6051896e7f3c">

So is there any particular reason why we create another `inline_response_200` record instead of directly creating the union type inline. Is there a way to improve this behaviour?

Reply:
Swagger parser treats this as an inline composed schema and flattens it into a named schema. Instead of using inline schemas, we could modify the YAML to use named schemas.